### PR TITLE
Bluetooth: controller: split: Port radio event abort

### PR DESCRIPTION
**Pull request by cvinayak**
Port the implementation that does radio event abort due to
ISR latencies. The implementation measures if the ISR could
not meet the hard real time deadline and closes the event
early.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c [hunk 1]
```diff
@@ -340,11 +341,25 @@ u32_t lll_evt_offset_get(struct evt_hdr *evt)
 u32_t lll_preempt_calc(struct evt_hdr *evt, u8_t ticker_id,
 		       u32_t ticks_at_event)
 {
-	/* TODO: */
+	u32_t ticks_now = ticker_ticks_now_get();
+	u32_t diff;
+
+	diff = ticker_ticks_diff_get(ticks_now, ticks_at_event);
+	diff += 3;
```
Comment (carlescufi) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c]: a small comment here about this hardcoded value?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c]: Help me here with a define name; 3 is the number of ticks between the high watermark and full tank. i.e. if the actual ticks plus 3 ticks exceeds the hard real-time deadline tick.

Comment (carlescufi) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c]: `SAFETY_TICKS` or `TICKS_TO_DEADLINE`

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c]: @carlescufi the right macro to use is HAL_TICKER_CNTR_CMP_OFFSET_MIN, hence updated now.

**Comment by cvinayak** [comment]
@thoh-ot @aescolar ...if the changes seems ok (although its nordic lll specific), please do the needful.

**Review by mtpr-ot — APPROVED** [review]
I don't see any issues here, but since this is mostly Nordic specific code (except for the getters), I'm not an expert on the impact. OK with me to merge, if accepted by Nordic.

**Comment by aescolar** [comment]
@cvinayak As @carlescufi is out, would it make sense if instead @joerchan would take a look to have somebody else from Nordic giving their blessing? Otherwise just tell and I press the button.

**Pull request merged**